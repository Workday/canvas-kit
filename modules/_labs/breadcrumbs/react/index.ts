import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
