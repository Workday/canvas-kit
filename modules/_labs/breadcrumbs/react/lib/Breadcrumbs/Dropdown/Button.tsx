/** @jsx jsx */
import React from 'react';
import {css, jsx} from '@emotion/core';
import {IconButton, IconButtonProps} from '@workday/canvas-kit-react-button';
import {folderCloseIcon} from '@workday/canvas-system-icons-web';
import {colors} from '@workday/canvas-kit-react-core';

interface DropdownButtonProps extends IconButtonProps {
  buttonRef: React.Ref<HTMLButtonElement>;
  setIsDropdownOpen: React.Dispatch<React.SetStateAction<boolean>>;
  toggled: boolean;
}

export const DropdownButton = ({
  buttonRef,
  setIsDropdownOpen,
  toggled,
  variant = IconButton.Variant.Plain,
  'aria-label': ariaLabel = 'More links',
  ...props
}: DropdownButtonProps) => {
  const hasPlainVariant = variant === IconButton.Variant.Plain;
  // gives the plain icon variant a little extra space between the chevron separators
  const customStyles = css(hasPlainVariant ? {margin: '0 1px'} : {});

  return (
    <IconButton
      css={customStyles}
      variant={variant}
      icon={folderCloseIcon}
      color={colors.licorice200}
      buttonRef={buttonRef}
      toggled={toggled}
      onClick={() => setIsDropdownOpen(!toggled)}
      aria-label={ariaLabel}
      aria-pressed={undefined}
      aria-expanded={toggled}
      aria-haspopup={true}
      aria-controls="menu"
      {...props}
    />
  );
};
