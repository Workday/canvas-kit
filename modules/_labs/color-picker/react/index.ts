export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
