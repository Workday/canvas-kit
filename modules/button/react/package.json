{
  "name": "@workday/canvas-kit-react-button",
  "version": "4.0.0-beta.5",
  "description": "A collection of various Canvas Buttons",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/Workday/canvas-kit/tree/master/modules/button/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "clean": "rimraf dist && rimraf ts3.5 && rimraf .build-info && mkdirp dist && mkdirp ts3.5/dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "button"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/is-prop-valid": "^0.8.2",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-labs-react-core": "4.0.0-beta.5",
    "@workday/canvas-kit-react-common": "4.0.0-beta.5",
    "@workday/canvas-kit-react-core": "4.0.0-beta.5",
    "@workday/canvas-kit-react-icon": "4.0.0-beta.5",
    "@workday/canvas-system-icons-web": "^1.0.20",
    "@workday/design-assets-types": "^0.2.4"
  }
}
