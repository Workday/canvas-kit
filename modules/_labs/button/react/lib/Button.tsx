import * as React from 'react';
import {colors} from '@workday/canvas-kit-react-core';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {GrowthBehavior} from '@workday/canvas-kit-react-common';
import {ButtonVariant, ButtonSize, ButtonColorCollection} from './types';
import {ButtonContainer, ButtonLabel, ButtonLabelData, ButtonLabelIcon} from './parts';

export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement>, GrowthBehavior {
  /**
   * The variant of the Button.
   * @default ButtonVariant.Secondary
   */
  variant: ButtonVariant;
  /**
   * The size of the Button.
   * @default ButtonSize.Medium
   */
  size: ButtonSize;
  /**
   * The ref to the button that the styled component renders.
   */
  buttonRef?: React.Ref<HTMLButtonElement>;
  /**
   * The data label of the Button.
   */
  dataLabel?: String;
  /**
   * The icon of the Button.
   */
  icon?: CanvasSystemIcon;
}

const Button = (props: ButtonProps) => {
  const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = props;

  return (
    // eslint-disable-next-line no-use-before-define
    <ButtonContainer colors={buttonColors[variant]} size={size} ref={buttonRef} {...elemProps}>
      {icon && <ButtonLabelIcon size={size} icon={icon} />}
      <ButtonLabel size={size} variant={variant}>
        {children}
      </ButtonLabel>
      {dataLabel && <ButtonLabelData size={size}>{dataLabel}</ButtonLabelData>}
    </ButtonContainer>
  );
};

Button.Variant = ButtonVariant;
Button.Size = ButtonSize;

Button.defaultProps = {
  variant: ButtonVariant.Secondary,
  size: ButtonSize.Medium,
};

export default Button;

export const buttonColors: ButtonColorCollection = {
  [ButtonVariant.Primary]: {
    default: {
      background: colors.blueberry400,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    hover: {
      background: colors.blueberry500,
      border: 'transparent',
      label: colors.frenchVanilla100,
      icon: colors.frenchVanilla100,
    },
    active: {
      background: colors.blueberry600,
      border: 'transparent',
      label: colors.frenchVanilla100,
      icon: colors.frenchVanilla100,
    },
    focus: {
      background: colors.blueberry400,
      border: 'transparent',
      label: colors.frenchVanilla100,
      icon: colors.frenchVanilla100,
    },
    disabled: {
      background: colors.blueberry200,
      border: 'transparent',
      label: colors.frenchVanilla100,
      icon: colors.frenchVanilla100,
    },
  },
  [ButtonVariant.Secondary]: {
    default: {
      background: colors.soap200,
      border: 'transparent',
      icon: colors.licorice200,
      label: colors.blackPepper400,
      labelData: colors.blackPepper400,
    },
    hover: {
      background: colors.soap400,
      border: 'transparent',
      label: colors.blackPepper400,
      icon: colors.licorice500,
    },
    active: {
      background: colors.soap500,
      border: 'transparent',
      label: colors.blackPepper400,
      icon: colors.licorice500,
    },
    focus: {
      background: colors.soap200,
      border: 'transparent',
      label: colors.blackPepper400,
      icon: colors.licorice200,
    },
    disabled: {
      background: colors.soap100,
      border: 'transparent',
      label: colors.licorice100,
      icon: colors.soap600,
      labelData: colors.licorice100,
    },
  },
  [ButtonVariant.OutlinePrimary]: {
    default: {
      background: 'transparent',
      border: colors.blueberry400,
      icon: colors.blueberry400,
      label: colors.blueberry400,
    },
    hover: {
      background: colors.blueberry400,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    active: {
      background: colors.blueberry500,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    focus: {
      background: colors.blueberry400,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    disabled: {
      background: colors.frenchVanilla100,
      border: colors.soap500,
      icon: colors.soap600,
      label: colors.licorice100,
    },
  },
  [ButtonVariant.OutlineSecondary]: {
    default: {
      background: 'transparent',
      border: colors.soap500,
      icon: colors.licorice200,
      label: colors.blackPepper400,
    },
    hover: {
      background: colors.licorice500,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    active: {
      background: colors.licorice600,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    focus: {
      background: colors.licorice500,
      border: 'transparent',
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    disabled: {
      background: colors.frenchVanilla100,
      border: colors.soap500,
      icon: colors.soap600,
      label: colors.licorice100,
    },
  },
  [ButtonVariant.OutlineInverse]: {
    default: {
      background: 'transparent',
      border: colors.frenchVanilla100,
      icon: colors.frenchVanilla100,
      label: colors.frenchVanilla100,
    },
    hover: {
      background: colors.frenchVanilla100,
      border: 'transparent',
      icon: colors.licorice500,
      label: colors.blackPepper400,
      labelData: colors.licorice300,
    },
    active: {
      background: colors.soap300,
      border: 'transparent',
      icon: colors.licorice500,
      label: colors.blackPepper400,
      labelData: colors.licorice300,
    },
    focus: {
      background: colors.frenchVanilla100,
      border: 'transparent',
      icon: colors.licorice500,
      label: colors.blackPepper400,
      labelData: colors.licorice300,
      // TODO: focusRingInner: 'currentColor',
      // TODO: focusRingOuter: colors.frenchVanilla100,
    },
    disabled: {
      background: 'transparent',
      border: 'rgba(255, 255, 255, 0.75)',
      icon: 'rgba(255, 255, 255, 0.75)',
      label: 'rgba(255, 255, 255, 0.75)',
      labelData: 'rgba(255, 255, 255, 0.75)',
    },
  },
};
