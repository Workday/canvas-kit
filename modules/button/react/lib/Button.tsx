/** @jsx jsx */
import {jsx} from '@emotion/core';
import * as React from 'react';
import {ButtonBaseCon, ButtonBaseLabel, ButtonLabelData, ButtonLabelIcon} from './ButtonBase';
import {DeprecatedButtonVariant, ButtonSize, ButtonVariant} from './types';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {GrowthBehavior} from '@workday/canvas-kit-react-common';
import {labelDataBaseStyles} from './ButtonStyles';

export interface BaseButtonProps<T = ButtonVariant | DeprecatedButtonVariant>
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * The variant of the Button.
   * @default ButtonVariant.Secondary
   */
  variant?: T;
  /**
   * The size of the Button.
   * @default ButtonSize.Medium
   */
  size?: ButtonSize;
  /**
   * The ref to the button that the styled component renders.
   */
  buttonRef?: React.Ref<HTMLButtonElement>;
  /**
   * The data label of the Button.
   */
  dataLabel?: String;
  /**
   * The icon of the Button.
   */
  icon?: CanvasSystemIcon;
}

export interface ButtonProps<T = ButtonVariant | DeprecatedButtonVariant>
  extends BaseButtonProps<T>,
    GrowthBehavior {
  /**
   * The children of the Button (cannot be empty).
   */
  children: React.ReactNode;
}

export default class Button extends React.Component<ButtonProps> {
  public static Variant = ButtonVariant;
  public static Size = ButtonSize;

  static defaultProps = {
    size: ButtonSize.Medium,
    variant: ButtonVariant.Secondary,
    grow: false,
  };

  public render() {
    const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = this.props;

    // Restrict Hightlight button to only being sized Large, Medium with an Icon
    if (variant === ButtonVariant.Highlight && (icon === undefined || size === ButtonSize.Small)) {
      return null;
    }

    return (
      <ButtonBaseCon variant={variant} size={size} ref={buttonRef} {...elemProps}>
        {icon && <ButtonLabelIcon size={size} icon={icon} />}
        <ButtonBaseLabel size={size} variant={variant}>
          {children}
        </ButtonBaseLabel>
        {dataLabel && (
          <ButtonLabelData size={size} className={labelDataBaseStyles.classname}>
            {dataLabel}
          </ButtonLabelData>
        )}
      </ButtonBaseCon>
    );
  }
}
/**
 * @deprecated deprecated_Button in @workday/canvas-kit-react-button will be removed soon. Use Button instead.
 */
// eslint-disable-next-line @typescript-eslint/class-name-casing
export class deprecated_Button extends React.Component<ButtonProps<DeprecatedButtonVariant>> {
  public static Variant = DeprecatedButtonVariant;
  public static Size = ButtonSize;

  static defaultProps = {
    size: ButtonSize.Large,
    variant: DeprecatedButtonVariant.Secondary,
    grow: false,
  };

  public componentDidMount() {
    console.warn('This component is now deprecated, consider using the new Button component');
  }

  public render() {
    const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = this.props;

    return <Button {...this.props} {...elemProps} />;
  }
}
