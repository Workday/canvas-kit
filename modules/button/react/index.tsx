import Button from './lib/Button';

export default Button;
export {default as Button, beta_Button, ButtonProps} from './lib/Button';
export {default as IconButton, IconButtonProps} from './lib/IconButton';
export {default as DropdownButton} from './lib/DropdownButton';
export {default as TextButton, TextButtonProps} from './lib/TextButton';
export {
  default as IconButtonToggleGroup,
  IconButtonToggleGroupProps,
} from './lib/IconButtonToggleGroup';
export * from './lib/types';
