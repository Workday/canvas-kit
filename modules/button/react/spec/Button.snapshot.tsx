import * as React from 'react';
import Button from '../lib/Button';
import * as renderer from 'react-test-renderer';

describe('Button Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Button>Button</Button>);
    expect(component).toMatchSnapshot();
  });

  test('renders a button with no type defined', () => {
    const component = renderer.create(<Button variant={undefined}>Button</Button>);
    expect(component).toMatchSnapshot();
  });

  test('renders a growing button', () => {
    const component = renderer.create(<Button grow={true}>Button</Button>);
    expect(component).toMatchSnapshot();
  });
});
