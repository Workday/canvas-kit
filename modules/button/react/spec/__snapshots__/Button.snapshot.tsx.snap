// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots renders a button with no type defined 1`] = `
.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2 span .wd-icon-fill,
.emotion-2 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-2 .button-label-data {
  color: #333333;
}

.emotion-2:focus {
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
}

.emotion-2:focus span .wd-icon-fill,
.emotion-2:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover:focus {
  background-color: #dfe2e6;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ced3d9;
  border-color: transparent;
  color: #333333;
}

.emotion-2:active span .wd-icon-fill,
.emotion-2:focus:active span .wd-icon-fill,
.emotion-2:hover:active span .wd-icon-fill,
.emotion-2:active span .wd-icon-accent,
.emotion-2:focus:active span .wd-icon-accent,
.emotion-2:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:hover {
  background-color: #dfe2e6;
  border-color: transparent;
  color: #333333;
}

.emotion-2:hover span .wd-icon-fill,
.emotion-2:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #f6f7f8;
  border-color: transparent;
  color: #a1aab3;
}

.emotion-2:disabled span .wd-icon-fill,
.emotion-2:active:disabled span .wd-icon-fill,
.emotion-2:focus:disabled span .wd-icon-fill,
.emotion-2:hover:disabled span .wd-icon-fill,
.emotion-2:disabled span .wd-icon-accent,
.emotion-2:active:disabled span .wd-icon-accent,
.emotion-2:focus:disabled span .wd-icon-accent,
.emotion-2:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-2:disabled .button-label-data,
.emotion-2:active:disabled .button-label-data,
.emotion-2:focus:disabled .button-label-data,
.emotion-2:hover:disabled .button-label-data {
  color: #a1aab3;
}

.emotion-2:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

<button
  className="emotion-2 emotion-3"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
</button>
`;

exports[`Button Snapshots renders a growing button 1`] = `
.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
  width: 100%;
  max-width: 100%;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2 span .wd-icon-fill,
.emotion-2 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-2 .button-label-data {
  color: #333333;
}

.emotion-2:focus {
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
}

.emotion-2:focus span .wd-icon-fill,
.emotion-2:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover:focus {
  background-color: #dfe2e6;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ced3d9;
  border-color: transparent;
  color: #333333;
}

.emotion-2:active span .wd-icon-fill,
.emotion-2:focus:active span .wd-icon-fill,
.emotion-2:hover:active span .wd-icon-fill,
.emotion-2:active span .wd-icon-accent,
.emotion-2:focus:active span .wd-icon-accent,
.emotion-2:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:hover {
  background-color: #dfe2e6;
  border-color: transparent;
  color: #333333;
}

.emotion-2:hover span .wd-icon-fill,
.emotion-2:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #f6f7f8;
  border-color: transparent;
  color: #a1aab3;
}

.emotion-2:disabled span .wd-icon-fill,
.emotion-2:active:disabled span .wd-icon-fill,
.emotion-2:focus:disabled span .wd-icon-fill,
.emotion-2:hover:disabled span .wd-icon-fill,
.emotion-2:disabled span .wd-icon-accent,
.emotion-2:active:disabled span .wd-icon-accent,
.emotion-2:focus:disabled span .wd-icon-accent,
.emotion-2:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-2:disabled .button-label-data,
.emotion-2:active:disabled .button-label-data,
.emotion-2:focus:disabled .button-label-data,
.emotion-2:hover:disabled .button-label-data {
  color: #a1aab3;
}

.emotion-2:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

<button
  className="emotion-2 emotion-3"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
</button>
`;

exports[`Button Snapshots renders as expected 1`] = `
.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2 span .wd-icon-fill,
.emotion-2 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-2 .button-label-data {
  color: #333333;
}

.emotion-2:focus {
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
}

.emotion-2:focus span .wd-icon-fill,
.emotion-2:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover:focus {
  background-color: #dfe2e6;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ced3d9;
  border-color: transparent;
  color: #333333;
}

.emotion-2:active span .wd-icon-fill,
.emotion-2:focus:active span .wd-icon-fill,
.emotion-2:hover:active span .wd-icon-fill,
.emotion-2:active span .wd-icon-accent,
.emotion-2:focus:active span .wd-icon-accent,
.emotion-2:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:hover {
  background-color: #dfe2e6;
  border-color: transparent;
  color: #333333;
}

.emotion-2:hover span .wd-icon-fill,
.emotion-2:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #f6f7f8;
  border-color: transparent;
  color: #a1aab3;
}

.emotion-2:disabled span .wd-icon-fill,
.emotion-2:active:disabled span .wd-icon-fill,
.emotion-2:focus:disabled span .wd-icon-fill,
.emotion-2:hover:disabled span .wd-icon-fill,
.emotion-2:disabled span .wd-icon-accent,
.emotion-2:active:disabled span .wd-icon-accent,
.emotion-2:focus:disabled span .wd-icon-accent,
.emotion-2:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-2:disabled .button-label-data,
.emotion-2:active:disabled .button-label-data,
.emotion-2:focus:disabled .button-label-data,
.emotion-2:hover:disabled .button-label-data {
  color: #a1aab3;
}

.emotion-2:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

<button
  className="emotion-2 emotion-3"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
</button>
`;
